tools/github-membership/membership.go

```diff
@@ -0,0 +1,119 @@
+/*
+* Copyright 2023 Google LLC. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package membership
+
+import (
+	"fmt"
+	"math/rand"
+	"slices"
+	"time"
+
+	"golang.org/x/exp/maps"
+)
+
+type UserType int64
+
+const (
+	CommunityUserType UserType = iota
+	GooglerUserType
+	CoreContributorUserType
+)
+
+func (ut UserType) String() string {
+	switch ut {
+	case GooglerUserType:
+		return "Googler"
+	case CoreContributorUserType:
+		return "Core Contributor"
+	default:
+		return "Community Contributor"
+	}
+}
+
+func (gh *Client) GetUserType(user string) UserType {
```

Comment:
undefined: Clientcompiler[UndeclaredName](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#UndeclaredName)

Should this define a client interface that has this function?